File: 游戏引擎类算法/按钮组底层实现/buttonGroup.js
var lights = [0,0,0,0];
function change(idx)
{
	lights[idx] = 1;
	for(index=0; index < lights.length; index++)
	{
		if(index != idx)
		{
			lights[index] = 0;
		}
	}
	return lights;
}
console.log(change(3))

File: 游戏引擎类算法/按钮组底层实现/buttonGroup.c
#include <stdio.h>
int status[] = {0,0,0,0};
int *change(int idx)
{
	status[idx] = 1;
	int length = sizeof(status)/sizeof(int);
	for(int index=0; index < length; index++)
	{
		if(index != idx)
		{
			status[index] = 0;
		}
	}
	return status;
}
int main()
{
	int *p_result = change(3);
	for(int i=0; i< sizeof(status)/sizeof(int); i++)
	{
		printf("%d",p_result[i]);
	}
	return 0;
}